diff -dur ejabberd-0.7.orig/src/ejabberd.cfg.example ejabberd-0.7/src/ejabberd.cfg.example
--- ejabberd-0.7.orig/src/ejabberd.cfg.example	2004-07-11 22:51:50.000000000 +0200
+++ ejabberd-0.7/src/ejabberd.cfg.example	2004-09-01 11:59:00.267069546 +0200
@@ -1,5 +1,11 @@
 % $Id$
 
+% This is a setup for working "localhost" server.
+% To make fully functional world-accessible Jabber server you will need to:
+% - change 'localhost' everywhere below to your desired Jabber server domain name
+% - provide SSL certificates and uncomment ssl settings
+% - modify everything else according to your needs
+
 %override_acls.
 
 
@@ -24,8 +30,9 @@
 % Only admins can use configuration interface:
 {access, configure, [{allow, admin}]}.
 
-% Every username can be registered via in-band registration:
-{access, register, [{allow, all}]}.
+
+% In-band registration disabled, let the admin decide who will use his server
+{access, register, [{allow, none}]}.
 
 % After successful registration user will get message with following subject
 % and body:
@@ -92,12 +99,25 @@
 % Listened ports:
 {listen,
  [{5222, ejabberd_c2s,     [{access, c2s}, {shaper, c2s_shaper}]},
-  {5223, ejabberd_c2s,     [{access, c2s}, ssl, {certfile, "./ssl.pem"}]},
+
+% SSL disabled, enable _after_ providing the certificate
+% {5223, ejabberd_c2s,     [{access, c2s}, ssl, {certfile, "./ssl.pem"}]},
+  
   {5269, ejabberd_s2s_in,  [{shaper, s2s_shaper}]},
-  {5280, ejabberd_http,    [http_poll, web_admin]},
-  {8888, ejabberd_service, [{access, all},
-			    {hosts, ["icq.localhost", "sms.localhost"],
-			     [{password, "secret"}]}]}
+
+% External services. 
+% To enable more than one you _must_ change port numbers (here and in service's config)
+%  {5347, ejabberd_service, [{access, all},
+%			    {hosts, ["icq.localhost", "sms.localhost"],
+%			     [{password, ">secret<"}]}]},
+%  {5347, ejabberd_service, [{access, all},
+%			    {host, "gg.localhost",
+%			     [{password, ">secret<"}]}]},
+%  {5347, ejabberd_service, [{access, all},
+%			    {hosts, ["ircnet.localhost", "freenode.localhost"],
+%			     [{password, ">secret<"}]}]},
+
+  {5280, ejabberd_http,    [http_poll, web_admin]}
  ]}.
 
 % If SRV lookup fails, then port 5269 is used to communicate with remote server
