--- src/web/ejabberd_web_admin.erl	2008-05-20 12:22:03.000000000 +0200
+++ src/web/ejabberd_web_admin.erl-new	2008-07-16 20:54:04.518671250 +0200
@@ -1385,7 +1385,7 @@
 	     lists:map(
 	       fun(_SU = {Server, User}) ->
 		       US = {User, Server},
-		       QueueLen = length(mnesia:dirty_read({offline_msg, US})),
+		       QueueLen = 0,
 		       FQueueLen = [?AC(URLFunc({users_queue, Prefix,
 						 User, Server}),
 					integer_to_list(QueueLen))],
