diff -dur -x '*~' ejabberd-1.1.0.orig/src/Makefile.in ejabberd-1.1.0/src/Makefile.in
--- ejabberd-1.1.0.orig/src/Makefile.in	2006-04-01 18:06:36.000000000 +0200
+++ ejabberd-1.1.0/src/Makefile.in	2006-04-26 10:32:28.000000000 +0200
@@ -34,13 +34,13 @@
 
 DESTDIR = 
 
-EJABBERDDIR = $(DESTDIR)@prefix@/var/lib/ejabberd
+EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd
 BEAMDIR = $(EJABBERDDIR)/ebin
 PRIVDIR = $(EJABBERDDIR)/priv
 SODIR = $(PRIVDIR)/lib
 MSGSDIR = $(PRIVDIR)/msgs
-LOGDIR = $(DESTDIR)@prefix@/var/log/ejabberd
-ETCDIR = $(DESTDIR)@prefix@/etc/ejabberd
+LOGDIR = $(DESTDIR)/var/log/ejabberd
+ETCDIR = $(DESTDIR)/etc/jabber
 
 ASN_FLAGS = -bber_bin +der +compact_bit_string +optimize +noobj
 
