--- ejabberd-2.0.0/src/Makefile.in.orig	2008-02-21 11:02:12.000000000 +0100
+++ ejabberd-2.0.0/src/Makefile.in	2008-02-22 14:52:47.000000000 +0100
@@ -50,13 +50,13 @@
 
 DESTDIR =
 
-EJABBERDDIR = $(DESTDIR)@prefix@/var/lib/ejabberd
+EJABBERDDIR = $(DESTDIR)@libdir@/ejabberd
 BEAMDIR = $(EJABBERDDIR)/ebin
 PRIVDIR = $(EJABBERDDIR)/priv
 SODIR = $(PRIVDIR)/lib
 MSGSDIR = $(PRIVDIR)/msgs
-LOGDIR = $(DESTDIR)@prefix@/var/log/ejabberd
-ETCDIR = $(DESTDIR)@prefix@/etc/ejabberd
+LOGDIR = $(DESTDIR)/var/log/ejabberd
+ETCDIR = $(DESTDIR)/etc/jabber
 SBINDIR = $(DESTDIR)@prefix@/sbin
 
 ifeq ($(shell uname),Darwin)
