diff -urN ejabberd-14.07.org/ejabberdctl.template ejabberd-14.07/ejabberdctl.template
--- ejabberd-14.07.org/ejabberdctl.template	2014-07-22 17:42:49.000000000 +0200
+++ ejabberd-14.07/ejabberdctl.template	2014-10-15 09:38:33.692517899 +0200
@@ -8,6 +8,7 @@
 ERL_MAX_ETS_TABLES=1400
 FIREWALL_WINDOW=""
 ERLANG_NODE=ejabberd@localhost
+HOME=/home/services/jabber; export HOME
 
 # define default environment variables
 SCRIPT_DIR=`cd ${0%/*} && pwd`
@@ -28,7 +28,7 @@ if [ "$INSTALLUSER" != "" ] ; then
                 mkdir -p "$INSTALLUSER_HOME"
                 chown "$INSTALLUSER" "$INSTALLUSER_HOME"
             fi
-            EXEC_CMD="su $INSTALLUSER -c"
+            EXEC_CMD="su -s /bin/sh $INSTALLUSER -c"
         fi
     done
     if [ `id -g` -eq `id -g $INSTALLUSER` ] ; then
@@ -55,7 +56,7 @@
 
 # Define ejabberd variable if they have not been defined from the command line
 if [ "$ETC_DIR" = "" ] ; then
-    ETC_DIR={{sysconfdir}}/ejabberd
+    ETC_DIR={{sysconfdir}}/jabber
 fi
 if [ "$EJABBERDCTL_CONFIG_PATH" = "" ] ; then
     EJABBERDCTL_CONFIG_PATH=$ETC_DIR/ejabberdctl.cfg
@@ -103,7 +104,7 @@
 SASL_LOG_PATH=$LOGS_DIR/erlang.log
 DATETIME=`date "+%Y%m%d-%H%M%S"`
 ERL_CRASH_DUMP=$LOGS_DIR/erl_crash_$DATETIME.dump
-ERL_INETRC=$ETC_DIR/inetrc
+ERL_INETRC=$ETC_DIR/ejabberd-inetrc
 
 # define erl parameters
 ERLANG_OPTS="+K $POLL -smp $SMP +P $ERL_PROCESSES $ERL_OPTIONS"
diff -urN ejabberd-14.07.org/Makefile.in ejabberd-14.07/Makefile.in
--- ejabberd-14.07.org/Makefile.in	2014-07-22 17:42:49.000000000 +0200
+++ ejabberd-14.07/Makefile.in	2014-10-15 09:32:42.603846549 +0200
@@ -9,7 +9,7 @@
 DESTDIR =
 
 # /etc/ejabberd/
-ETCDIR = $(DESTDIR)@sysconfdir@/ejabberd
+ETCDIR = $(DESTDIR)@sysconfdir@/jabber
 
 # /sbin/
 SBINDIR = $(DESTDIR)@sbindir@
@@ -123,7 +123,7 @@
 	[ -f $(ETCDIR)/ejabberdctl.cfg ] \
 		&& $(INSTALL) -b -m 640 $(G_USER) ejabberdctl.cfg.example $(ETCDIR)/ejabberdctl.cfg-new \
 		|| $(INSTALL) -b -m 640 $(G_USER) ejabberdctl.cfg.example $(ETCDIR)/ejabberdctl.cfg
-	$(INSTALL) -b -m 644 $(G_USER) inetrc $(ETCDIR)/inetrc
+	$(INSTALL) -b -m 644 $(G_USER) inetrc $(ETCDIR)/ejabberd-inetrc
 	#
 	# Administration script
 	[ -d $(SBINDIR) ] || $(INSTALL) -d -m 755 $(SBINDIR)
