diff -dur ejabberd-13.10.orig/Makefile.in ejabberd-13.10/Makefile.in
--- ejabberd-13.10.orig/Makefile.in	2013-09-27 11:19:22.000000000 +0200
+++ ejabberd-13.10/Makefile.in	2013-11-14 09:23:06.000000000 +0100
@@ -9,7 +9,7 @@
 DESTDIR =
 
 # /etc/ejabberd/
-ETCDIR = $(DESTDIR)@sysconfdir@/ejabberd
+ETCDIR = $(DESTDIR)@sysconfdir@/jabber
 
 # /sbin/
 SBINDIR = $(DESTDIR)@sbindir@
Only in ejabberd-13.10: Makefile.in~
