diff -urNp -x '*.orig' ejabberd-21.01.org/Makefile.in ejabberd-21.01/Makefile.in
--- ejabberd-21.01.org/Makefile.in	2021-01-29 10:01:29.000000000 +0100
+++ ejabberd-21.01/Makefile.in	2021-03-03 14:25:43.980087766 +0100
@@ -9,7 +9,7 @@ exec_prefix = @exec_prefix@
 DESTDIR =
 
 # /etc/ejabberd/
-ETCDIR = @sysconfdir@/ejabberd
+ETCDIR = @sysconfdir@/jabber
 
 # /bin/
 BINDIR = @bindir@
@@ -318,7 +318,7 @@ install: copy-files ejabberdctl.example
 	[ -f $(DESTDIR)$(ETCDIR)/ejabberdctl.cfg ] \
 		&& $(INSTALL) -b -m 640 $(G_USER) ejabberdctl.cfg.example $(DESTDIR)$(ETCDIR)/ejabberdctl.cfg-new \
 		|| $(INSTALL) -b -m 640 $(G_USER) ejabberdctl.cfg.example $(DESTDIR)$(ETCDIR)/ejabberdctl.cfg
-	$(INSTALL) -b -m 644 $(G_USER) inetrc $(DESTDIR)$(ETCDIR)/inetrc
+	$(INSTALL) -b -m 644 $(G_USER) inetrc $(DESTDIR)$(ETCDIR)/ejabberd-inetrc
 	#
 	# Administration script
 	[ -d $(DESTDIR)$(SBINDIR) ] || $(INSTALL) -d -m 755 $(DESTDIR)$(SBINDIR)
diff -urNp -x '*.orig' ejabberd-21.01.org/ejabberdctl.template ejabberd-21.01/ejabberdctl.template
--- ejabberd-21.01.org/ejabberdctl.template	2021-01-29 10:01:29.000000000 +0100
+++ ejabberd-21.01/ejabberdctl.template	2021-03-03 14:25:43.973420891 +0100
@@ -9,6 +9,7 @@ ERL_MAX_ETS_TABLES=1400
 FIREWALL_WINDOW=""
 INET_DIST_INTERFACE=""
 ERLANG_NODE=ejabberd@localhost
+HOME=/home/services/jabber; export HOME
 
 # define default environment variables
 [ -z "$SCRIPT" ] && SCRIPT=$0
@@ -86,7 +86,7 @@ fi
 [ -f "$VMARGS" ] && ERLANG_OPTS="$ERLANG_OPTS -args_file $VMARGS"
 ERL_LIBS='{{libdir}}'
 ERL_CRASH_DUMP="$LOGS_DIR"/erl_crash_$(date "+%Y%m%d-%H%M%S").dump
-ERL_INETRC="$CONFIG_DIR"/inetrc
+ERL_INETRC="$CONFIG_DIR"/ejabberd-inetrc
 
 # define ejabberd parameters
 EJABBERD_OPTS="$EJABBERD_OPTS\
