diff -ur ejabberd/src/Makefile.in ejabberd.new/src/Makefile.in
--- ejabberd/src/Makefile.in	2008-07-22 20:46:21.436869256 +0200
+++ ejabberd.new/src/Makefile.in	2008-07-22 20:54:06.653072763 +0200
@@ -74,7 +74,7 @@
 DESTDIR =
 
 # /etc/ejabberd/
-ETCDIR = $(DESTDIR)@sysconfdir@/ejabberd
+ETCDIR = $(DESTDIR)@sysconfdir@/jabber
 
 # /sbin/
 SBINDIR = $(DESTDIR)@sbindir@
@@ -85,7 +85,7 @@
 # /share/doc/ejabberd
 PACKAGE_TARNAME = @PACKAGE_TARNAME@
 datarootdir = @datarootdir@
-DOCDIR = @docdir@
+DOCDIR = $(DESTDIR)@docdir@
 
 # /usr/lib/ejabberd/ebin/
 BEAMDIR = $(EJABBERDDIR)/ebin
Tylko w ejabberd.new/src: Makefile.in~
