--- ejabberd-18.09/ejabberd.yml.example~	2018-09-26 15:01:27.000000000 +0200
+++ ejabberd-18.09/ejabberd.yml.example	2018-12-05 10:30:31.241701930 +0100
@@ -1,6 +1,13 @@
 ###
 ###              ejabberd configuration file
 ###
+
+### This is a setup for working "localhost" server.
+### To make fully functional world-accessible Jabber server you will need to:
+### - change 'localhost' everywhere below to your desired Jabber server domain name
+### - provide SSL certificates and uncomment ssl settings
+### - modify everything else according to your needs
+
 ### The parameters used in this configuration file are explained at
 ###
 ###       https://docs.ejabberd.im/admin/configuration
