From 9a8245b8e6a17b306d07cd96f2483044dd4981d2 Mon Sep 17 00:00:00 2001
From: Evgeny Khramtsov <ekhramtsov@process-one.net>
Date: Wed, 4 Sep 2019 10:31:09 +0300
Subject: [PATCH] Fix the regression introduced in dd57950

Fixes #3010
---
 src/mod_ping.erl | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/mod_ping.erl b/src/mod_ping.erl
index d4c226e56a..8e6827247f 100644
--- a/src/mod_ping.erl
+++ b/src/mod_ping.erl
@@ -132,8 +132,6 @@ handle_cast(Msg, State) ->
     ?WARNING_MSG("Unexpected cast: ~p", [Msg]),
     {noreply, State}.
 
-handle_info({iq_reply, #iq{type = error}, JID}, State) ->
-    handle_info({iq_reply, timeout, JID}, State);
 handle_info({iq_reply, #iq{}, _JID}, State) ->
     {noreply, State};
 handle_info({iq_reply, timeout, JID}, State) ->
